controller/index.js

```diff
@@ -1,1019 +1,1054 @@
-        if ( req.params.format ){
+            agol.getItem(data.host, req.params.item, req.query, function (err, itemJson) {
+              if (err) {
+                return res.status(500).send(err)
+              }
 
-          // force an override on the format param if given a format in the query
-          if ( req.query.format ){
-            req.params.format = req.query.format;
-            delete req.query.format;
-          }
+              if (exists) {
+                return agol.isExpired(info, req.query.layer, function (err, isExpired) {
+                  fileParams.err = err
+                  if (!isExpired) {
+                    return controller._returnFile(req, res, path, fileParams.fileName)
+                  }
 
-          // redirect to thumbnail for png access
-          if (req.params.format == 'png'){
-            controller.thumbnail(req, res);
-          } else {
+                  // if it's expired, then remove the data and request a new file
+                  agol.dropItem(data.host, req.params.item, req.query, function () {
+                    req.query.format = req.params.format
+                    controller._getItemData(req, res, function (err, itemJson) {
+                      fileParams.err = err
+                      fileParams.itemJson = itemJson
+                      fileParams.data = (itemJson && itemJson.data && itemJson.data[0]) ? itemJson.data[0] : null
+                      // var used to request new files if needed.
+                      controller._requestNewFile(fileParams)
+                    })
+                  })
+                })
+              }
 
-            // change geojson to json
-            req.params.format = req.params.format.replace('geojson', 'json');
-            // use the item as the file dir so we can organize exports by id
-            var dir = req.params.item + '_' + ( req.params.layer || 0 );
-            var path = [ 'files', dir, key ].join( '/' );
-            // the file name for the export 
-            var name = ( info && info.info ) ? info.name || info.info.name || info.info.title : key;
-            name = (name.length > 150) ? name.substr(0, 150): name;
-            var fileName = name + '.' + req.params.format;
-            fileName = fileName.replace(/\/|,|&|\|/g, '').replace(/ /g, '_').replace(/\(|\)/g, '');
-
-            // if we have a layer then append it to the query params 
-            if ( req.params.layer ) {
-              req.query.layer = req.params.layer;
+              req.query.format = req.params.format
+              controller._getItemData(req, res, function (err, itemJson) {
+                fileParams.err = err
+                fileParams.itemJson = itemJson
+                fileParams.data = (itemJson && itemJson.data && itemJson.data[0]) ? itemJson.data[0] : null
+                controller._requestNewFile(fileParams)
+              })
+            })
+          })
+        })
+      } else {
```

Comment:
wooooah that's a big block of logic inside a single if. I think the above file processing stream should be broken out into it's own private method